packages/library-legacy-sham/src/__tests__/public-key-test.ts

```diff
@@ -0,0 +1,106 @@
+        expect(() =>
+            // This is basically just complaining that `default` is not callable.
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            PublicKey[method]()
+        ).toThrow(`PublicKey#${method.toString()} is unimplemented`);
+    });
+    it('vends the all-zero address at `default`', () => {
+        expect(PublicKey.default.toBase58()).toBe('11111111111111111111111111111111');
+    });
+    describe('given a valid public key', () => {
+        let publicKey: PublicKey;
+        beforeEach(() => {
+            publicKey = new PublicKey(VALID_ADDRESS);
+        });
+        describe('the `encode()` method', () => {
+            if (__NODEJS__) {
+                it('returns a buffer representing the public key bytes', () => {
+                    expect(publicKey.encode()).toEqual(Buffer.from(VALID_ADDRESS_BYTES));
+                });
+            } else {
+                it('fatals', () => {
+                    expect(() => {
+                        publicKey.encode();
+                    }).toThrow();
+                });
+            }
+        });
+        describe('the `equals()` method', () => {
+            it('returns `true` given two public keys with the same address', () => {
+                expect(publicKey.equals(publicKey)).toBe(true);
+            });
+            it('returns `false` given two public keys with the same address', () => {
+                const publicKeyA = new PublicKey('11111111111111111111111111111111');
+                const publicKeyB = new PublicKey('11111111111111111111111111111112');
+                expect(publicKeyA.equals(publicKeyB)).toBe(false);
+            });
+        });
+        describe('the `toBase58()` method', () => {
+            it('returns the `Address` related to the public key', () => {
+                expect(publicKey.toBase58()).toBe(VALID_ADDRESS);
+            });
+        });
+        describe('the `toBuffer()` method', () => {
+            if (__NODEJS__) {
+                it('returns a buffer representing the public key bytes', () => {
+                    expect(publicKey.toBuffer()).toEqual(Buffer.from(VALID_ADDRESS_BYTES));
+                });
+            } else {
+                it('fatals', () => {
+                    expect(() => {
+                        publicKey.toBuffer();
+                    }).toThrow();
+                });
+            }
+        });
+        describe('the `toBytes()` method', () => {
+            it('returns a byte array representation of the public key', () => {
+                expect(publicKey.toBytes()).toEqual(new Uint8Array(VALID_ADDRESS_BYTES));
+            });
+        });
+        describe('the `toJSON()` method', () => {
+            it('returns the `Address` related to the public key as a string', () => {
+                expect(publicKey.toJSON()).toBe(VALID_ADDRESS);
```

Comment:
I may be wrong here but shouldn't `toJson()` return a JSON string instead of just a string? I.e. `"I'm valid JSON"` instead of `I'm not`.

```suggestion
                expect(publicKey.toJSON()).toBe(`"${VALID_ADDRESS}"`);
```

Reply:
This confused me too in something else where I was using `JSON.stringify(..)` but it seems like JavaScript ignores the quotes.

Reply:
Oh I see, `toJSON` doesn't return you a JSON string? Just the JSON-compatible JavaScript object?

Reply:
The more I think about it the less sure I am, but I'm imagining it's a JSON string until you pass it into a JavaScript function, then they are clipped?

Reply:
I'm just matching the behaviour of the existing library without trying to use my brain. I might hurt myself otherwise.